### ISSUE TITLE
Path math needs special handling for query (?) string

### ISSUE THREAD

**Issue by peller** [maqetta/maqetta#2348]
for example, if you append /mydir to http://maqetta.org/? it should go prior to the ? -- http://maqetta.org/mydir?

Currently, the option to close files in the workspace puts a ? on the url like this, and requests for helpers fail

**Comment by JonFerraiolo** [comment]
-> M6 High. (Wasn't there already a bug regarding the `?` ? I couldn't find it, though)

**Comment by jhpedemonte** [comment]
Another solution: redirect from `foo.com/maqetta/?resetWorkspace` (once it has been handled) back to the main page (i.e. `foo.com/maqetta/`).  Could do this for all such URL params.

**Comment by childsb** [comment]
to recreate this bug, just browse to ./maqetta/? and see what fails to load.

**Comment by jhpedemonte** [comment]
-> @aerwin, can you take a look at this one?

**Comment by aerwin** [comment]
> -> @aerwin, can you take a look at this one?

Sure... looking at it I'm not sure I fully appreciate the issue or the severity...

E.g., I can go to http://localhost:8081/maqetta/? and see some things don't load. But, with @childsb 's commit from 3 months ago, do we still have a Prio High/Sev 2 issue? What other scenarios would cause there to be a "/?" in the URL in the first place?

**Comment by JonFerraiolo** [comment]
Some history first:

As I remember, the issue with `http://localhost:8081/maqetta/?` came up because after a user entered `foo.com/maqetta/?resetWorkspace` we were redirecting to `foo.com/maqetta/?` and things wouldn't work, and users would have to manually remove the `?` for Maqetta to appear. At the time, this was deemed to be a bad situation because occasionally users get stuck because the hand-edit their HTML file, they get the RSOD, and they had no way to tell Maqetta to quit opening their current editor. So we invented `foo.com/maqetta/?resetWorkspaceState=true` but originally that redirected to `foo.com/maqetta/?` instead of `foo.com/maqetta/`, and then we had to tell users to take one additional manual step. And most of the time, these instructions were happening on the maqetta-users forum, sometimes with hours delay response between the steps.

But I just tested Preview6 on maqetta.org and yesterday's build of M7, and the server redirects from `foo.com/maqetta/?resetWorkspaceState=true` to `foo.com/maqetta/`, so it seems to me we could make this a Priority-Low bug. (as @aerwin suggests). 

What I would suggest is that @aerwin lower the priority, and as a result it will almost certainly get pushed to M8, and wait and see if anyone objects to this priority changing.

**Comment by peller** [comment]
I agree it is no longer critical and a fix should not be attempted for M7, but I'd prefer going back to prio-med so that it is more likely to get addressed.   Severity could certainly be lowered, too, but it's a flaw in logic that could come back to bite us.

**Comment by aerwin** [comment]
--> Prio Med, Sev 3, M8
